lib/support.mli

```diff
@@ -1,7 +1,25 @@
 (** Support module for recently introduced function in stdlib *)
 
 val extension: string -> string
+(** [extension name] is the shortest suffix [ext] of [name0] where:
+    - [name0] is the longest suffix of [name] that does not contain
+      a directory separator;
+    - [ext] does {b not} starts with a period (such as
+      [name0 ^ "." ^ ext = name])
```

Comment:
The two conditions `shortest suffix` and `does not start with a period` do not mix well: at this point the empty word still satisfies the two conditions. I would reword it as:
```
- [.] [ext] is a suffix of [name0]
```